glitter/pages/admin.py

```diff
@@ -71,5 +77,64 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.grid_view)),
             url(r'^tree/$', wrap(self.changelist_view), name='%s_%s_tree' % info),
+            url(r'^(\d+)/duplicate/$', wrap(self.duplicate_page), name='%s_%s_duplicate' % info),
         ] + urlpatterns
         return urlpatterns
+
+    def duplicate_page(self, request, obj_id):
+        obj = get_object_or_404(Page, id=obj_id)
+
+        if request.method == "POST":
+            form = DuplicatePageForm(request.POST or None)
+            if form.is_valid():
+                new_page = form.save()
+
+                # Use current version if exists if not get the latest
+                if obj.current_version:
+                    current_version = obj.current_version
+                else:
+                    current_version = obj.get_latest_version()
+
+                # Create a new version
+                new_version = Version(content_object=new_page)
+                new_version.template_name = current_version.template_name
+                new_version.version_number = 1
+                new_version.owner = request.user
+                new_version.save()
+
+                #  Copy all blocks.
+                for block in current_version.contentblock_set.all():
```

Comment:
Need to use `glitter.utils.duplicate` to copy the actual blocks first. Have a look at https://github.com/blancltd/django-glitter/blob/202084450a4a2528a8dc3ee26ba8e5c531038e39/glitter/admin.py#L273.

Otherwise two page versions will have content blocks pointing to the same block.

Perhaps this could be made as a more generic method/function? Up to you!